### PR TITLE
feat: add Pinterest time-series line chart echart

### PR DESCRIPTION
**Pull request by kgopal492**
### SUMMARY

Create a Pinterest time-series line chart visualization that copies uses all of the features of the current line chart visualization, but replaces the tooltip with a "delta table" of D/D, W/W, M/M, etc. data.

* Can select which delta values are shown in tooltip delta table
Screenshot 2023-02-13 at 5 56 47 PM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Pinterest/Line/tooltip.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        param.seriesName,
+      );
+      const newRow = this.deltaTableColumns.map(column => {
+        const columnData = deltaTableData[column];
+        let color;
+        let data = columnData ?? 'N/A';
+        if (column === DeltaTableColumn.METRIC) {
+          data = param.marker + escape(columnData);
+        }
+        if (PERCENT_CHANGE_COLUMNS.includes(column) && columnData != null) {
```
Comment (suryabhaskarkarri) [inline superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Pinterest/Line/tooltip.ts:174]: else if

**Review by suryabhaskarkarri — COMMENTED** [review]
All the code is new (or) part of it is copied from our existing delta table component ?

**Comment by kgopal492** [comment]
@suryabhaskarkarri Nothing's copied from the delta table component, but `superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Pinterest/Line/index.ts` is copied from the existing time-series line chart.

File: superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Pinterest/Line/tooltip.ts [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        let params: CallbackDataParams[] = richTooltip
+          ? (initialParams as CallbackDataParams[])
+          : [initialParams as CallbackDataParams];
+        if (tooltipSortByMetric) {
+          params = orderBy(params, [
+            ({ value }) => -1 * value[yIndex],
+            ['desc'],
+          ]) as CallbackDataParams[];
+        }
+        const deltaTableRows = this.getDeltaTableRows(params, xIndex);
```
Comment (suryabhaskarkarri) [inline superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Pinterest/Line/tooltip.ts:228]: even if richtooltip is false, you are using your custom formatter to display tooltip. In that case can it be defaulted to superset default tooltip formatter (or) that is not possible ?

Comment (suryabhaskarkarri) [reply superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/Pinterest/Line/tooltip.ts:228]: just saw that delta columns can be configured even for non-rich-tooltip

**Review by suryabhaskarkarri — APPROVED** [review]
rest of the code looks good to me.  There are lint failures, please address them before merging

**Pull request merged**